master/buildbot/process/buildstep.py

```diff
@@ -566,6 +560,14 @@ def setRenderable(res, attr):
             # we describe ourselves only when renderables are interpolated
             self.realUpdateSummary()
 
+            # check doStepIf (after rendering)
```

Comment:
I agree on putting it after the rendering, but converting str/unicode to bool shouldn't be necessary since the renderable should return bool and nothing else beside that. The last case is just to convert the result from the callable to something usable.
A unit test which fails before the changes and passes after would be much appreciated